### PR TITLE
✨ Redis Client and ConnectionCache

### PR DESCRIPTION
**Pull request by RTLS**

### CODE REVIEW

**Comment by RTLS** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/557?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#557** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/557?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/557?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#556** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/556?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#555** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/555?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: lib/sequin/application.ex [hunk 1]
```diff
@@ -56,7 +55,8 @@ defmodule Sequin.Application do
       {ConCache, name: Sequin.Cache, ttl_check_interval: :timer.seconds(1), global_ttl: :infinity},
       {Oban, Application.fetch_env!(:sequin, Oban)},
       {Redix, Application.fetch_env!(:redix, :start_opts)},
-      ConnectionCache,
+      Sequin.Databases.ConnectionCache,
+      Sequin.Redis.ConnectionCache,
```
Comment (acco) [inline lib/sequin/application.ex:59]: 👀

File: lib/sequin/consumers/redis_destination.ex [hunk 2]
```diff
@@ -7,26 +7,25 @@ defmodule Sequin.Consumers.RedisDestination do
...
+  @derive {Jason.Encoder, only: [:host, :port, :key]}
   typed_embedded_schema do
     field :type, Ecto.Enum, values: [:redis], default: :redis
     field :host, :string
     field :port, :integer
     field :username, :string
     field :password, Sequin.Encrypted.Binary
     field :tls, :boolean, default: false
-    field :key_prefix, :string
+    field :key, :string
```
Comment (acco) [inline lib/sequin/consumers/redis_destination.ex:18]: ah right - because we're just `xadd`'ing to one key

**Review by acco — APPROVED** [review]
🕺 🕺 🕺

**Pull request merged**